src/analysis/timeline/timelineCredView.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+
+import {sum} from "d3-array";
+import sortBy from "lodash.sortby";
+import stringify from "json-stable-stringify";
+import * as NullUtil from "../../util/null";
+import {TimelineCred, type CredNode} from "./timelineCred";
+import {type Interval, weekIntervals} from "./interval";
+import {type NodeAddressT, NodeAddress} from "../../core/graph";
+
+// test intervals in timelineCredView subsset of the
+// intervals from the base TimelineCred
+export class TimelineCredView {
+  _inputInterval: Interval;
+  _viewIntervals: Interval[];
+  _timelineCred: TimelineCred;
+
+  // We are displaying a range of the intervals in the base TimelineCred
+  // We track the start index and end index of the range, so we can efficiently
+  // slice the cred
+  _startIntervalIndex: number;
+  _endIntervalIndex: number;
+
+  constructor(timelineCred: TimelineCred, inputInterval: Interval) {
+    this._timelineCred = timelineCred;
+    this._inputInterval = inputInterval;
+    this._viewIntervals = weekIntervals(
+      inputInterval.startTimeMs,
+      inputInterval.endTimeMs
+    );
+    this._findIntervalIndices();
+  }
+
+  _findIntervalIndices() {
+    const baseIntervalStrings = new Set(
+      this._timelineCred.intervals().map(stringify)
+    );
+    const viewIntervalStrings = this._viewIntervals.map(stringify);
+    for (const v of viewIntervalStrings) {
+      if (!baseIntervalStrings.has(v)) {
+        throw new Error(`invariant violation: can't find interval ${v}`);
+      }
+    }
+    this._startIntervalIndex = Array.from(baseIntervalStrings).indexOf(
+      viewIntervalStrings[0]
+    );
+    this._endIntervalIndex = Array.from(baseIntervalStrings).indexOf(
+      viewIntervalStrings[viewIntervalStrings.length - 1]
+    );
+  }
+
+  intervals(): $ReadOnlyArray<Interval> {
+    return this._viewIntervals;
+  }
+
+  /**
+   * Get the CredNode for a given NodeAddress.
+   *
+   * Returns undefined if the node is not in the filtered results.
+   *
+   * Note that it's possible that the node is present in the Graph, but not the
+   * filtered results; if so, it will return undefined.
+   */
+  credNode(a: NodeAddressT): ?CredNode {
+    const fullView = this._timelineCred._cred.addressToCred.get(a);
+    if (fullView === undefined) {
+      return undefined;
+    }
+    const cred = fullView.slice(
+      this._startIntervalIndex,
+      this._endIntervalIndex
+    );
+    const total = sum(cred);
+    const node = NullUtil.get(this._timelineCred._graph.node(a));
+    return {cred, total, node};
+  }
+
+  /**
+   * Return all the nodes matching the prefix, along with their cred,
+   * sorted by total cred (descending).
+   */
+  credSortedNodesView(prefix: NodeAddressT): $ReadOnlyArray<CredNode> {
```

Comment:
Why the `View` in `credSortedNodesView`? I think `credSortedNodes` would suffice?

I think we could also go for simpler names: `node` and `nodes`. That it is a credNode is communicated by the types, and implied by the class we're calling the methods on. That the nodes are sorted by descending cred can also be communicated by the docstring. But in general I'm OK with more verbose/descriptive names too. Use your judgement. :)